Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\frontend\admin.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\crime\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /favicon.ico
Track cases view accessed by user: xyz1 (Regular User)
Raw SQL count of cases for user 5: 0
Found 0 cases for user 5
Statistics calculated - Total: 0, Active: 0, Resolved: 0
Context data - Total cases: 0, Active: 0, Resolved: 0
Number of cases in context: 0
Track cases view accessed by user: xyz1 (Regular User)
Raw SQL count of cases for user 5: 1
Found 1 cases for user 5
Case details - Number: MP-D79C60B4, Status: pending, Created: 2025-05-25 13:55:17.167054+00:00
Case photo URL: /media/missing_persons/screencapture-127-0-0-1-1445-2025-05-18-09_44_08.png
Statistics calculated - Total: 1, Active: 1, Resolved: 0
Found 0 user updates for case MP-D79C60B4
Found 0 officer updates for case MP-D79C60B4
Total updates for case MP-D79C60B4: 1
Context data - Total cases: 1, Active: 1, Resolved: 0
Number of cases in context: 1
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /cadmin/admin-dashboard/
Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 21, in admin_dashboard
    return render(request, 'admin_dashboard.html')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'admin_case_details' with no arguments not found. 1 pattern(s) tried: ['cadmin/case/(?P<case_id>[0-9]+)/\\Z']
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
Forbidden (CSRF token from POST incorrect.): /login/
Internal Server Error: /cadmin/admin-dashboard/
Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 21, in admin_dashboard
    return render(request, 'admin_dashboard.html')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'admin_case_details' with no arguments not found. 1 pattern(s) tried: ['cadmin/case/(?P<case_id>[0-9]+)/\\Z']
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /favicon.ico
Track cases view accessed by user: xyz (Regular User)
Raw SQL count of cases for user 2: 5
Found 5 cases for user 2
Case details - Number: MP-E26954F1, Status: pending, Created: 2025-05-25 12:14:10.463259+00:00
Case photo URL: /media/missing_persons/ali.jpg
Case details - Number: MP-07FDE01A, Status: pending, Created: 2025-05-25 12:01:44.832311+00:00
Case photo URL: /media/missing_persons/Untitled_design_7.png
Case details - Number: MP-08567438, Status: pending, Created: 2025-05-25 09:55:37.098423+00:00
Case photo URL: /media/missing_persons/a31484e2ea4fcf8bcc8c7b7bf6160ea8.jpg
Case details - Number: MP-29D16945, Status: pending, Created: 2025-05-25 09:51:30.332683+00:00
Case photo URL: /media/missing_persons/images_6.jfif
Case details - Number: MP-77D05C4A, Status: pending, Created: 2025-05-25 09:45:55.237668+00:00
Case photo URL: /media/missing_persons/b17b8ef13714d5908e53f279ba8b5e5b.jpg
Statistics calculated - Total: 5, Active: 5, Resolved: 0
Found 0 user updates for case MP-E26954F1
Found 0 officer updates for case MP-E26954F1
Total updates for case MP-E26954F1: 1
Found 0 user updates for case MP-07FDE01A
Found 0 officer updates for case MP-07FDE01A
Total updates for case MP-07FDE01A: 1
Found 0 user updates for case MP-08567438
Found 0 officer updates for case MP-08567438
Total updates for case MP-08567438: 1
Found 0 user updates for case MP-29D16945
Found 0 officer updates for case MP-29D16945
Total updates for case MP-29D16945: 1
Found 0 user updates for case MP-77D05C4A
Found 0 officer updates for case MP-77D05C4A
Total updates for case MP-77D05C4A: 1
Context data - Total cases: 5, Active: 5, Resolved: 0
Number of cases in context: 5
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\frontend\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\frontend\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\frontend\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
User aabb1122 logged in
User type: admin
Is admin: True
Redirecting to admin dashboard
Watching for file changes with StatReloader
Not Found: /favicon.ico
Not Found: /missing-persons.html
User aabb1122 logged in
User type: admin
Is admin: True
Redirecting to admin dashboard
User aabb1122 logged in
User type: admin
Is admin: True
Redirecting to admin dashboard
Error in manage_cases view: Unsupported lookup 'date' for DateField or join on the field not permitted.
Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 98, in manage_cases
    cases = cases.filter(last_seen_date__date=today.date())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\query.py", line 1491, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\query.py", line 1509, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\query.py", line 1516, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1643, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1675, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1585, in build_filter
    condition = self.build_lookup(lookups, col, value)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1406, in build_lookup
    lhs = self.try_transform(lhs, lookup_name)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1455, in try_transform
    raise FieldError(
django.core.exceptions.FieldError: Unsupported lookup 'date' for DateField or join on the field not permitted.
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
Error in register_officer view: Cannot resolve keyword 'cnic' into field. Choices are: address, assigned_cases, assigned_missing_persons, caseevidence, created_at, created_cases, date_joined, date_of_birth, email, first_name, groups, id, is_active, is_staff, is_superuser, is_verified, last_login, last_name, logentry, officer_case_updates, password, phone_number, profile_picture, reported_missing_persons, updated_at, user_case_updates, user_permissions, user_type, username
Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 98, in register_officer
    if User.objects.filter(cnic=cnic).exists():
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\query.py", line 1491, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\query.py", line 1509, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\query.py", line 1516, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1643, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1675, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1523, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1330, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\db\models\sql\query.py", line 1802, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'cnic' into field. Choices are: address, assigned_cases, assigned_missing_persons, caseevidence, created_at, created_cases, date_joined, date_of_birth, email, first_name, groups, id, is_active, is_staff, is_superuser, is_verified, last_login, last_name, logentry, officer_case_updates, password, phone_number, profile_picture, reported_missing_persons, updated_at, user_case_updates, user_permissions, user_type, username
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\frontend\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\frontend\models.py changed, reloading.
Watching for file changes with StatReloader
Raw query: SELECT "frontend_user"."id", "frontend_user"."password", "frontend_user"."last_login", "frontend_user"."is_superuser", "frontend_user"."username", "frontend_user"."first_name", "frontend_user"."last_name", "frontend_user"."email", "frontend_user"."is_staff", "frontend_user"."is_active", "frontend_user"."date_joined", "frontend_user"."user_type", "frontend_user"."phone_number", "frontend_user"."address", "frontend_user"."profile_picture", "frontend_user"."date_of_birth", "frontend_user"."is_verified", "frontend_user"."created_at", "frontend_user"."updated_at", "frontend_user"."cnic", "frontend_user"."rank" FROM "frontend_user"
Number of users found: 7
Final statistics - Total: 7, Active: 7, Inactive: 0
Users in current page:
User: abc1, Email: abc1@gmail.com, Type: officer
User: xyz2, Email: xyz2@gmail.com, Type: user
User: xyz1, Email: xyz1@gmail.com, Type: user
User: aabb1122, Email: aabb1122@gmail.com, Type: admin
User: aabbcc, Email: aabbcc11@gmail.com, Type: admin
User: xyz, Email: xyz@gmail.com, Type: user
User: abcde, Email: abcde@gmail.com, Type: user
Raw query: SELECT "frontend_user"."id", "frontend_user"."password", "frontend_user"."last_login", "frontend_user"."is_superuser", "frontend_user"."username", "frontend_user"."first_name", "frontend_user"."last_name", "frontend_user"."email", "frontend_user"."is_staff", "frontend_user"."is_active", "frontend_user"."date_joined", "frontend_user"."user_type", "frontend_user"."phone_number", "frontend_user"."address", "frontend_user"."profile_picture", "frontend_user"."date_of_birth", "frontend_user"."is_verified", "frontend_user"."created_at", "frontend_user"."updated_at", "frontend_user"."cnic", "frontend_user"."rank" FROM "frontend_user"
Number of users found: 8
Final statistics - Total: 8, Active: 8, Inactive: 0
Users in current page:
User: abc2, Email: abc2@gmail.com, Type: officer
User: abc1, Email: abc1@gmail.com, Type: officer
User: xyz2, Email: xyz2@gmail.com, Type: user
User: xyz1, Email: xyz1@gmail.com, Type: user
User: aabb1122, Email: aabb1122@gmail.com, Type: admin
User: aabbcc, Email: aabbcc11@gmail.com, Type: admin
User: xyz, Email: xyz@gmail.com, Type: user
User: abcde, Email: abcde@gmail.com, Type: user
Raw query: SELECT "frontend_user"."id", "frontend_user"."password", "frontend_user"."last_login", "frontend_user"."is_superuser", "frontend_user"."username", "frontend_user"."first_name", "frontend_user"."last_name", "frontend_user"."email", "frontend_user"."is_staff", "frontend_user"."is_active", "frontend_user"."date_joined", "frontend_user"."user_type", "frontend_user"."phone_number", "frontend_user"."address", "frontend_user"."profile_picture", "frontend_user"."date_of_birth", "frontend_user"."is_verified", "frontend_user"."created_at", "frontend_user"."updated_at", "frontend_user"."cnic", "frontend_user"."rank" FROM "frontend_user"
Number of users found: 8
Final statistics - Total: 8, Active: 7, Inactive: 1
Users in current page:
User: abc2, Email: abc2@gmail.com, Type: officer
User: abc1, Email: abc1@gmail.com, Type: officer
User: xyz2, Email: xyz2@gmail.com, Type: user
User: xyz1, Email: xyz1@gmail.com, Type: user
User: aabb1122, Email: aabb1122@gmail.com, Type: admin
User: aabbcc, Email: aabbcc11@gmail.com, Type: admin
User: xyz, Email: xyz@gmail.com, Type: user
User: abcde, Email: abcde@gmail.com, Type: user
Raw query: SELECT "frontend_user"."id", "frontend_user"."password", "frontend_user"."last_login", "frontend_user"."is_superuser", "frontend_user"."username", "frontend_user"."first_name", "frontend_user"."last_name", "frontend_user"."email", "frontend_user"."is_staff", "frontend_user"."is_active", "frontend_user"."date_joined", "frontend_user"."user_type", "frontend_user"."phone_number", "frontend_user"."address", "frontend_user"."profile_picture", "frontend_user"."date_of_birth", "frontend_user"."is_verified", "frontend_user"."created_at", "frontend_user"."updated_at", "frontend_user"."cnic", "frontend_user"."rank" FROM "frontend_user"
Number of users found: 8
Final statistics - Total: 8, Active: 6, Inactive: 2
Users in current page:
User: abc2, Email: abc2@gmail.com, Type: officer
User: abc1, Email: abc1@gmail.com, Type: officer
User: xyz2, Email: xyz2@gmail.com, Type: user
User: xyz1, Email: xyz1@gmail.com, Type: user
User: aabb1122, Email: aabb1122@gmail.com, Type: admin
User: aabbcc, Email: aabbcc11@gmail.com, Type: admin
User: xyz, Email: xyz@gmail.com, Type: user
User: abcde, Email: abcde@gmail.com, Type: admin
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /cadmin/manage-user/
Not Found: /cadmin/manage-user/
Not Found: /cadmin/manage-user/
Not Found: /cadmin/manage-user/
C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py changed, reloading.
Watching for file changes with StatReloader
Error in admin_dashboard view: Reverse for 'manage_user' not found. 'manage_user' is not a valid view function or pattern name.
Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 49, in admin_dashboard
    return render(request, 'admin_dashboard.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'manage_user' not found. 'manage_user' is not a valid view function or pattern name.
Internal Server Error: /cadmin/admin-dashboard/
Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 49, in admin_dashboard
    return render(request, 'admin_dashboard.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'manage_user' not found. 'manage_user' is not a valid view function or pattern name.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 53, in admin_dashboard
    return render(request, 'admin_dashboard.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'manage_user' not found. 'manage_user' is not a valid view function or pattern name.
Error in admin_dashboard view: Reverse for 'manage_user' not found. 'manage_user' is not a valid view function or pattern name.
Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 49, in admin_dashboard
    return render(request, 'admin_dashboard.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'manage_user' not found. 'manage_user' is not a valid view function or pattern name.
Internal Server Error: /cadmin/admin-dashboard/
Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 49, in admin_dashboard
    return render(request, 'admin_dashboard.html', context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'manage_user' not found. 'manage_user' is not a valid view function or pattern name.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\contrib\auth\decorators.py", line 59, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\Desktop\mp\crime\cadmin\views.py", line 53, in admin_dashboard
    return render(request, 'admin_dashboard.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Abdul Qadoos Bhatti\AppData\Local\Programs\Python\Python312\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'manage_user' not found. 'manage_user' is not a valid view function or pattern name.
